extensions/Lily/Hitboxes.js

```diff
@@ -0,0 +1,93 @@
+(function (Scratch) {
+  "use strict";
+
+  const vm = Scratch.vm;
+  const runtime = vm.runtime;
+  const renderer = vm.renderer;
+
+  class Hitboxes {
+    constructor() {
+      runtime.on("BEFORE_EXECUTE", () => {
+        runtime.targets.forEach((target) => {
+          if (typeof target.hitboxSkin === "undefined") return;
+          const drawableID = target.drawableID;
+          const drawable = renderer._allDrawables[drawableID];
+          target.originalSkin = drawable.skin;
+          drawable.skin = target.hitboxSkin;
+        });
+      });
+      runtime.on("AFTER_EXECUTE", () => {
+        runtime.targets.forEach((target) => {
+          if (typeof target.originalSkin === "undefined") return;
+          const drawableID = target.drawableID;
+          const drawable = renderer._allDrawables[drawableID];
+          drawable.skin = target.originalSkin;
```

Comment:
```suggestion
          if (target.hitboxSkin === drawable.skin) drawable.skin = target.originalSkin;
```
There's a bug where trying to change costumes while a hitbox was active would cause weird behavior. This somewhat fixes the problem, however the hitbox will still behave weirdly if you change a costume and then try to sense if that costume is touching another within the same thread.